### DISCUSSION TITLE
[Config Support]: FFMPEG Errors After Switching to GO2RTC with Annke NCD800

### DISCUSSION THREAD

**Discussion by ianwood** [blakeblackshear/frigate#10041] [Config Support]
### Describe the problem you are having

Running into an FFMPEG issue after updating config to use GO2RTC with one of my two cameras, the Annke NCD800. My other camera, the Annke AC800 works fine in the new config. I suspect this is related to the resolution of the NCD800 being 5120 x 1440 (as discussed here: [Issue 6678](https://github.com/blakeblackshear/frigate/issues/6678)) but even after switching to H.265, it still occurs. Some more observations:

- Same camera and resolution using H.264 codec did not have this issue prior to updating config to use GO2RTC. I've included FFMPEG changes which may be relevant below. 
- Occurs with camera set to H.264 and H.265. See below errors and VLC screenshots. 
- GO2RTC has no problem with the resolution using either codec. Used VLC to confirm. 

### Version

0.13.2

### Frigate config file

```yaml
mqtt:
  host: core-mosquitto
  user: [------]
  password: [------]

environment_vars:
  LIBVA_DRIVER_NAME: i965

ffmpeg:
  hwaccel_args: preset-vaapi

go2rtc:
  streams:
    rear:
      - rtsp://[------]:[------]@rear-ipcam.home.lab/channels/101
      - isapi://[------]:[------]@rear-ipcam.home.lab:80/
    front:
      - rtsp://[------]:[------]@front-ipcam.home.lab/channels/101
      - isapi://[------]:[------]@front-ipcam.home.lab:80/

detectors:
  coral:
    type: edgetpu
    device: usb

cameras:
  rear:
    ffmpeg:
      inputs:
        - path: rtsp://[------]:[------]@rear-ipcam.home.lab/channels/102
          roles:
            - detect
        - path: rtsp://[------].home.lab:8554/rear
          roles:
            - record
    detect:
      width: 1280
      height: 720
      fps: 6
      annotation_offset: -900
    motion:
      mask:
      - 1280,318,1045,146,1026,84,982,80,980,0,1280,0
      - 0,0,0,296,373,181,960,179,955,0
      - 1245,683,1245,705,895,705,895,683
    snapshots:
      enabled: True
      timestamp: False
      bounding_box: True
      required_zones:
        - backyard
    record:
      enabled: True
      retain:
        days: 4
        mode: motion
      events:
        retain:
          default: 4
          mode: active_objects
        required_zones:
          - backyard
    zones:
      backyard:
        coordinates: 1162,362,1280,720,56,720,0,633,0,410,333,254,975,296,980,232,1087,238
        objects:
        - person

  front:
    ffmpeg:
      inputs:
        - path: rtsp://[------]:[------]@front-ipcam.home.lab/channels/102
          roles:
            - detect
        - path: rtsp://[------].home.lab:8554/front
          roles:
            - record
    detect:
      width: 1920
      height: 536
      fps: 6
      annotation_offset: -900
    motion:
      mask:
      - 1337,32,958,0,684,49,314,222,457,536,372,536,245,309,125,367,76,536,0,536,0,0,1920,0,1920,301
      - 1156,536,1096,459,983,455,879,393,644,436,609,536
    snapshots:
      enabled: True
      timestamp: False
      bounding_box: True
      required_zones:
        - frontyard
    record:
      enabled: True
      retain:
        days: 4
        mode: motion
      events:
        retain:
          default: 4
          mode: active_objects
        required_zones:
          - frontyard
    zones:
      frontyard:
        coordinates: 1471,201,662,176,416,281,408,432,85,472,77,536,1920,536
        objects:
        - person
```

### Relevant log output

```shell
With H.264 from camera:

2024-02-25 23:10:24.288407912  [2024-02-25 15:10:24] watchdog.front                 ERROR   : Ffmpeg process crashed unexpectedly for front.
2024-02-25 23:10:24.291713483  [2024-02-25 15:10:24] watchdog.front                 ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2024-02-25 23:10:24.291733808  [2024-02-25 15:10:24] ffmpeg.front.detect            ERROR   : [h264 @ 0x56033a073c40] Hardware does not support image size 5120x1440 (constraints: width 0-4096 height 0-4096).
2024-02-25 23:10:24.291740270  [2024-02-25 15:10:24] ffmpeg.front.detect            ERROR   : [h264 @ 0x56033a073c40] Failed setup for format vaapi: hwaccel initialisation returned error.
2024-02-25 23:10:24.291746495  [2024-02-25 15:10:24] ffmpeg.front.detect            ERROR   : Impossible to convert between the formats supported by the filter 'Parsed_fps_0' and the filter 'auto_scale_0'
2024-02-25 23:10:24.291825205  [2024-02-25 15:10:24] ffmpeg.front.detect            ERROR   : Error reinitializing filters!
2024-02-25 23:10:24.291909777  [2024-02-25 15:10:24] ffmpeg.front.detect            ERROR   : Failed to inject frame into filter network: Function not implemented
2024-02-25 23:10:24.291915777  [2024-02-25 15:10:24] ffmpeg.front.detect            ERROR   : Error while processing the decoded data for stream #0:0
2024-02-25 23:10:26.316698000  [2024-02-25 15:10:26] frigate.video                  ERROR   : front: Unable to read frames from ffmpeg process.
2024-02-25 23:10:26.318531480  [2024-02-25 15:10:26] frigate.video                  ERROR   : front: ffmpeg process is not running. exiting capture thread...

With H.265 from camera:

2024-02-25 23:29:17.304446447  [2024-02-25 15:29:17] watchdog.front                 ERROR   : Ffmpeg process crashed unexpectedly for front.
2024-02-25 23:29:17.304459434  [2024-02-25 15:29:17] watchdog.front                 ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2024-02-25 23:29:17.304719739  [2024-02-25 15:29:17] ffmpeg.front.detect            ERROR   : [hevc @ 0x55e7274060c0] Hardware does not support image size 5120x1440 (constraints: width 0-4096 height 0-4096).
2024-02-25 23:29:17.305565814  [2024-02-25 15:29:17] ffmpeg.front.detect            ERROR   : [hevc @ 0x55e7274060c0] Failed setup for format vaapi: hwaccel initialisation returned error.
2024-02-25 23:29:17.306452601  [2024-02-25 15:29:17] ffmpeg.front.detect            ERROR   : Impossible to convert between the formats supported by the filter 'Parsed_fps_0' and the filter 'auto_scale_0'
2024-02-25 23:29:17.307198005  [2024-02-25 15:29:17] ffmpeg.front.detect            ERROR   : Error reinitializing filters!
2024-02-25 23:29:17.308037881  [2024-02-25 15:29:17] ffmpeg.front.detect            ERROR   : Failed to inject frame into filter network: Function not implemented
2024-02-25 23:29:17.311803560  [2024-02-25 15:29:17] ffmpeg.front.detect            ERROR   : Error while processing the decoded data for stream #0:0
2024-02-25 23:29:19.795907997  [2024-02-25 15:29:19] frigate.video                  ERROR   : front: Unable to read frames from ffmpeg process.
2024-02-25 23:29:19.796775310  [2024-02-25 15:29:19] frigate.video                  ERROR   : front: Unable to read frames from ffmpeg process.
2024-02-25 23:29:19.798306353  [2024-02-25 15:29:19] frigate.video                  ERROR   : front: ffmpeg process is not running.
```

### Frigate stats

```json
FFPROBE OUTPUT
Stream 0:
Return Code: 0

Video:

Codec: H.265 / HEVC (High Efficiency Video Coding)
Resolution: 5120x1440
FPS: 15/1

Audio:

Codec: AAC (Advanced Audio Coding)

Stream 1:
Return Code: 0

Video:

Codec: H.265 / HEVC (High Efficiency Video Coding)
Resolution: 5120x1440
FPS: 15/1

Audio:

Codec: AAC (Advanced Audio Coding)

VAINFO OUTPUT
Return Code: 0

Process Output:

vainfo: VA-API version: 1.17 (libva 2.10.0)
vainfo: Driver version: Intel i965 driver for Intel(R) CherryView - 2.4.1
vainfo: Supported profile and entrypoints
VAProfileMPEG2Simple : VAEntrypointVLD
VAProfileMPEG2Simple : VAEntrypointEncSlice
VAProfileMPEG2Main : VAEntrypointVLD
VAProfileMPEG2Main : VAEntrypointEncSlice
VAProfileH264ConstrainedBaseline: VAEntrypointVLD
VAProfileH264ConstrainedBaseline: VAEntrypointEncSlice
VAProfileH264Main : VAEntrypointVLD
VAProfileH264Main : VAEntrypointEncSlice
VAProfileH264High : VAEntrypointVLD
```

### Operating system

HassOS

### Install method

HassOS Addon

### Coral version

USB

### Any other information that may be helpful

VLC confirmation of GO2RTC streams:

![2024-02-25-front-go2rtc-h264](https://github.com/blakeblackshear/frigate/assets/1890435/185895d2-e346-4005-a761-a2275396b035)
![2024-02-25-front-go2rtc-h265](https://github.com/blakeblackshear/frigate/assets/1890435/f07db94f-a046-4c90-8620-91dec8fe669c)

Prior config that worked had these settings for FFPMEG and the rstp stream with the camera set to H.264: 

ffmpeg:
  hwaccel_args: preset-vaapi
  output_args:
    record: -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c:v copy -c:a aac
    rtmp: -c:v copy -c:a aac -f flv

  front:
    ffmpeg:
      inputs:
        - path: rtsp://[------]:[------]@front-ipcam.home.lab:554/H264/ch1/sub/av_stream
          roles:
            - detect
        - path: rtsp://[------]:[------]@front-ipcam.home.lab:554/H264/ch1/main/av_stream
          roles:
            - record

**Comment by NickM-27** [comment]
I have this camera but the hikvision version. The problem is that your hardware doesn't support that high of a resolution. you'll need to use the sub stream for detection

**Comment by ianwood** [comment]
I am using the sub stream for detection. And it was working with the same hardware and full resolution using this config: 

ffmpeg:
hwaccel_args: preset-vaapi
output_args:
record: -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c:v copy -c:a aac
rtmp: -c:v copy -c:a aac -f flv

front:
ffmpeg:
inputs:
- path: rtsp://[------]:[------]@front-ipcam.home.lab:554/H264/ch1/sub/av_stream
roles:
- detect
- path: rtsp://[------]:[------]@front-ipcam.home.lab:554/H264/ch1/main/av_stream
roles:
- record

**Comment by NickM-27** [comment]
Those aren't correct urls for annke / hikvision. 

The sub stream would be `rtsp://[------]:[------]@rear-ipcam.home.lab/channels/102`

**Comment by ianwood** [comment]
Yes, if you look at my current config in the first post above, you'll see that is what I am using currently and it isn't working. 

But the one from my previous config (the incorrect URLs as you stated) actually did work.

**Comment by NickM-27** [comment]
Your logs show that you're are pulling 5120x1440, this is not the resolution of the sub stream. The ffprobe is interesting, seems this camera may be buggy or you just need to use the other url. 

My hikvision works without issues (but my hardware also supports decoding 5120x1440 hevc

**Comment by ianwood** [comment]
This Annke supports HEVC as well. And GO2RTC works perfectly with both as you can see from the VLC screenshots I posted above. 

Detect should be using: rtsp://[------]:[------]@front-ipcam.home.lab/channels/102

which is what is in the config. When I pull that up in VLC I get:

**Reply by NickM-27** [reply]
I am not talking about Annke, your server does not support decoding 5120x1440 whether it is h264 or h265. Looks like your screenshot is missing but your vlc screenshots above show both streams as 5120x1440 which is wrong for the sub stream.

**Comment by ianwood** [comment]
Sorry, I pulled the image because I didn't hide user/pass. The server did support 5120 x 1440 up until I changed the config. If I put back the old config, it works in H.264 at 5120 x 1440. 

This is the substream in current config. The resolution is not 5120 x 1440. It is the substream. 

![2024-02-25-front-substream](https://github.com/blakeblackshear/frigate/assets/1890435/4203e77f-e8bb-4798-ac9b-8c347fcaca31)

Also, I did notice the H.265 substream was choppy and dropping frames and have switched back to the H.264 substream. 

H.265 substream dropping frames. 
![2024-02-25-front-substream-h265-dropping](https://github.com/blakeblackshear/frigate/assets/1890435/491da902-856c-4355-88f3-b04811146054)

H.264 substream clean. 
![2024-02-25-front-substream-h264-clean](https://github.com/blakeblackshear/frigate/assets/1890435/9f1f3b8f-5bd0-4e2e-bdf4-db8f3527e25d)

**Reply by NickM-27** [reply]
Vaapi doesn't support 5120x1440 at h264 (in fact, neither does Nvidia decode). They both max out at 4096 in either dimension. H265 supports greater dimensions.

**Comment by ianwood** [comment]
If I use this: 

```ffmpeg:
  hwaccel_args: preset-vaapi
  output_args:
    record: -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c:v copy -c:a aac
    rtmp: -c:v copy -c:a aac -f flv

rtmp:
  enabled: False
```
And this 

```
  front:
    ffmpeg:
      inputs:
        - path: rtsp://[------]:[------]@front-ipcam.home.lab:554/H264/ch1/sub/av_stream
          roles:
            - detect
        - path: rtsp://[------]:[------]@front-ipcam.home.lab:554/H264/ch1/main/av_stream
          roles:
            - record
```

It works. But is that because I am not using VAAPI on that camera? Both cameras seem to be using the same amount of CPU. 
<img width="1182" alt="image" src="https://github.com/blakeblackshear/frigate/assets/1890435/67e2c594-c906-4411-a69a-3bc54bdc3729">

**Reply by ianwood** [reply]
It's weird because it was failing both H.265 and H.264 on the new config despite GO2RTC not having any issues. And on the old config it IS working in H.264 at full resolution.

**Reply by ianwood** [reply]
UPDATE: Changing the detect stream from: 

```rtsp://[------]:[------]@front-ipcam.home.lab:554/H264/ch1/sub/av_stream```

to

```rtsp://[------]:[------]@front-ipcam.home.lab/channels/102```

is what breaks it.

**Reply by ianwood** [reply]
Also, it looks like for whatever reason

```rtsp://[------]:[------]@front-ipcam.home.lab/channels/102```

is causing frigate to detect from the main stream at full resolution even though when I use the same rtsp link in VLC, it pulls up the correct sub stream. It was doing the same on my rear camera as well so I have reverted both detect streams to the "incorrect" format of: 

```rtsp://[------]:[------]@front-ipcam.home.lab:554/H264/ch1/sub/av_stream```

for the time being.

**Reply by NickM-27** [reply]
This is most likely a bug in the camera, potentially because vlc uses udp by default meanwhile frigate uses tcp by default. This behavior does not exist on my hikvision variant of this camera.

**Reply by ianwood** [reply]
I've just noticed the Annke AC800 has the same behavior as the NCD800 when using 

```rtsp://[user]:[pass]@[camera-ip]/channels/102``` 

for the detect sub-stream. It seems to get redirected to the main stream. And this AC800 camera has been flashed with Hikvision FW.
